src/main/java/com/google/sps/data/Section.java

```diff
@@ -15,33 +15,76 @@
 package com.google.sps.data;
 
 import java.util.Collections;
+import java.util.LinkedList;
 import java.util.List;
 
 public class Section {
-    private String professor;
-    // Meeting times will be sorted in ascending order of time, 
-    // with the first lesson being first in the list
-    private List<TimeRange> meetingTimes;
-
-    public Section(String professor, List<TimeRange> meetingTimes){
-        this.professor = professor;
-        this.meetingTimes = meetingTimes;
-        Collections.sort(this.meetingTimes, TimeRange.ORDER_BY_START);
-
-        for(int i = 0; i < meetingTimes.size() - 1; i++){
-          if(meetingTimes.get(i).overlaps(meetingTimes.get(i + 1))) {
-            throw new IllegalArgumentException("A section's lecture times can't overlap!");
-          }
-        }
+  private String professor;
+  // Meeting times will be sorted in ascending order of time,
+  // with the first lesson being first in the list
+  private List<TimeRange> meetingTimes;
+
+  public Section(String professor, List<TimeRange> meetingTimes) {
+    this.professor = professor;
+    this.meetingTimes = meetingTimes;
+    Collections.sort(this.meetingTimes, TimeRange.ORDER_BY_START);
+
+    for (int i = 0; i < meetingTimes.size() - 1; i++) {
+      if (meetingTimes.get(i).overlaps(meetingTimes.get(i + 1))) {
+        throw new IllegalArgumentException("A section's lecture times can't overlap!");
+      }
     }
+  }
+
+  public List<TimeRange> getMeetingTimes() {
+    return this.meetingTimes;
+  }
+
+  public String getProfessor() {
+    return this.professor;
+  }
+
+  /**
+   * Function to return whether or not a section overlaps with another.
+   * 
+   * @param other The other section.
+   * @return If there's overlap between the current section and the other section
+   */
+  public boolean overlaps(Section other) {
+    LinkedList<TimeRange> thisTimes = new LinkedList<>(this.meetingTimes);
+    LinkedList<TimeRange> otherTimes = new LinkedList<>(other.getMeetingTimes());
+
+    TimeRange thisCurr = thisTimes.removeFirst();
+    TimeRange otherCurr = otherTimes.removeFirst();
 
-    /**
-     * Function to return whether or not a section overlaps with another
-     * @param other
-     * @return If there's overlap between the current section and the other section
-     */
-    public boolean overlaps(Section other){
-        //TODO: This
+    while (!thisTimes.isEmpty() && !otherTimes.isEmpty()) {
```

Comment:
This will not end up checking for all possible collisions. If you have a `thisTimes` with times, `A, B` and a `otherTimes` with time `C` like so:

```
A |--|
B       |----|
C        |----|
```

You'll take the first element of both lists, `A` and `C`, get a `false` when checking the condition for this while loop, then go on to the last `if` check between `A` and `C`, which will pass since they don't overlap. `B` and `C` will not have been tested here.

The edge case here is that if you end up hitting the end of one of the lists of `TimeRange`s, you stop checking for overlaps between the last element of the list you reached the end of and the other remaining items in the other list.

I would encourage you to write a unit test for this case to validate that it does indeed break, then to write the code that fixes that test.